src/PostHog/Config/PostHogOptions.cs

```diff
@@ -33,6 +33,26 @@ public sealed class PostHogOptions : IOptions<PostHogOptions>
     /// </summary>
     public Uri HostUrl { get; set; } = new("https://us.i.posthog.com");
 
+    /// <summary>
+    /// Timeout in milliseconds for any calls. Defaults to 10 seconds.
+    /// </summary>
+    public TimeSpan RequestTimeout { get; set; } = TimeSpan.FromSeconds(10);
+
+    /// <summary>
+    /// Number of times to retry a failed request. Defaults to 3.
+    /// </summary>
+    public int MaxRetries { get; set; } = 3;
```

Comment:
We should probably validate this (and the other new settings) in the `PostHogClient` ctor.

For example, we may want to have a max value on the allowed retries. Make sure its not negative. Etc.

Same thing for the timespans. Make sure they're not less than `TimeSpan.Zero`.  Etc.